backend/api/cms/tests/factories.py

```diff
@@ -92,3 +92,18 @@ class GenericPageFactory(PageFactory):
 
     class Meta:
         model = GenericPage
+
+
+@register
+class SiteFactory(SiteFactory):
+    """
+    Overrides wagtail_factories.SiteFactory to use "testserver" as hostname
+    to make sure it works with Wagtail's ALLOWED_HOSTS in test environments.
+    """
+
+    hostname = "testserver"
```

Comment:
I would leave this anyway as this fixture is more specifc aka links the GenericPage as root_pag and use `testserver` as hostname that is the allowed hosts configured by wagtail in the tests I discovered.